crates/agglayer-contracts/src/aggchain.rs

```diff
@@ -1,16 +1,38 @@
 use ethers::{providers::Middleware, types::Address};
+use sp1_primitives::consts::{bytes_to_words_le, words_to_bytes_le};
 use tracing::error;
 
 use crate::{aggchain_base::AggchainBase, L1RpcClient, L1RpcError};
 
+#[derive(PartialEq, Eq)]
+pub struct AggchainVkey([u8; 32]);
```

Comment:
I think this would be a more accurate name:
```suggestion
pub struct AggchainVkeyHash([u8; 32]);
```